### PR TITLE
Adding += and + to Swift Dictionary

### PR DESCRIPTION
**Pull request by codecaffeine**
```swift
var foo = ["Hello": 1234]
let bar = ["World": 5678]
foo += bar // ["Hello": 1234, "World": 5678]
```

Note, if the keys are the same, the RHS key overrides the LHS key

### CODE REVIEW

**Comment by KVTaniguchi** [comment]
thank you

File: Pod/Classes/DictionaryConvenience.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Matt Thomas on 5/20/16.
+//
+//
+
+import Foundation
+
+public func += <Key, Value> (inout left: Dictionary<Key, Value>, right: Dictionary<Key, Value>) {
+    for (key, value) in right {
+        left.updateValue(value, forKey: key)
```
Comment (ndis1) [inline Pod/Classes/DictionaryConvenience.swift]: you prefer this syntax to the classic `[key] = value` ?

**Comment by KVTaniguchi** [comment]
conflicts

File: Pod/Classes/DictionaryConvenience.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//  DictionaryConvenience.swift
+//  Pods
+//
+//  Created by Matt Thomas on 5/20/16.
+//
+//
+
+import Foundation
+
+public func += <Key, Value> (inout left: [Key: Value], right: [Key: Value]) {
```
Comment (jgrandelli) [inline Pod/Classes/DictionaryConvenience.swift]: Is it worth making these parameters optional? Or at least the left one?

Comment (codecaffeine) [inline Pod/Classes/DictionaryConvenience.swift]: That's an interesting idea. I also want to add something to this PR that I use in Restock (`+` operator) so hold off on this PR at the moment.

Comment (codecaffeine) [inline Pod/Classes/DictionaryConvenience.swift]: I attempted to make these parameters optional and when I tried to use the operator, I got errors saying I needed to unwrap the optional use use it with the operator.

*I* don’t have any needs for optional parameters. Maybe when somebody does, they can investigate this.

**Comment by codecaffeine** [comment]
This PR also removes the `Example/Pods/` directory from source control since it was causing conflicts each time it was used (hence the -900 deleted lines)

**Comment by codecaffeine** [comment]
Added nil RHS to operators

**Comment by codecaffeine** [comment]
I attempted to do the nil for the LHS of the operators, but ran into issues with the compiler not being able to apply them.

**Pull request merged**